src/gordon_gcp/plugins/enricher.py

```diff
@@ -175,9 +174,16 @@ def __init__(self, config, http_client, success_channel, error_channel):
         instance_data = await self._poll_for_instance_data(
             event_data['resourceName'], msg_logger)
 
-        a_record = await self._create_A_rrecord(event_data, instance_data,
-                                                msg_logger)
-        return [a_record]
+        dns_zone = self.config['dns_zone']
+        hostname = event_data['resourceName'].split('/')[-1]
+        fqdn = '.'.join([hostname, dns_zone])
+
+        rrecords = []
+        for rrecord_creator in self._rrecord_creators:
```

Comment:
This might be a limited-scope optimization since it only allows a subclass to create more records within the same zone. If I were writing a subclass, I would be more inclined to override this whole `_create_rrecords` method, so I could have more flexibility both to use more zones and for the way in which I could design the class.